app/services/login/user_acceptable_verified_credential_updater_logger.rb

```diff
@@ -0,0 +1,174 @@
+
+      increment_statsd
+      log_info
+    end
+
+    private
+
+    attr_reader :user_avc
+
+    def increment_statsd
+      statsd_keys.each do |key|
+        StatsD.increment(key, 1)
+      end
+    end
+
+    def log_info
+      Rails.logger.info(LOG_MESSAGE, log_payload)
+    end
+
+    def statsd_keys
+      @statsd_keys ||= build_statsd_keys
+    end
+
+    def log_payload
+      @log_payload ||= build_log_payload
+    end
+
+    def added_type
+      @added_type ||= if avc_added?
+                        AVC_TYPE
+                      elsif ivc_added?
+                        IVC_TYPE
+                      end
+    end
+
+    def added_from_type
+      @added_from_type ||= if from_mhv?
+                             MHV_TYPE
+                           elsif from_dslogon?
+                             DSLOGON_TYPE
+                           elsif from_logingov?
+                             LOGINGOV_TYPE
+                           elsif from_idme?
+                             IDME_TYPE
+                           end
+    end
+
+    def build_statsd_keys
+      keys = []
+      return keys unless added_type.present? && added_from_type.present?
+
+      keys << "#{STATSD_KEY_PREFIX}.#{added_from_type}.#{added_type}.added"
+
+      if added_from_type == MHV_TYPE || added_from_type == DSLOGON_TYPE
+        keys << "#{STATSD_KEY_PREFIX}.#{MHV_TYPE}_#{DSLOGON_TYPE}.#{added_type}.added"
+      end
+
+      keys
+    end
+
+    def build_log_payload
+      payload = {}
+      payload[:added_type] = added_type
+      payload[:added_from] = added_from_type if added_from_type.present?
+      payload[:user_account_id] = user_account.id
+      payload[:mhv_uuid] = mhv_credential.mhv_uuid if added_from_type == MHV_TYPE
+      if added_from_type == DSLOGON_TYPE
+        payload[:dslogon_uuid] =
+          dslogon_credential.dslogon_uuid
+      end
+      payload[:idme_uuid] = idme_credential&.idme_uuid
+      payload[:logingov_uuid] = logingov_credential&.logingov_uuid
+
+      payload
+    end
+
+    def user_account
+      @user_account ||= user_avc.user_account
+    end
+
+    def idme_credential
+      @idme_credential ||= user_verifications.idme.first
+    end
+
+    def logingov_credential
+      @logingov_credential ||= user_verifications.logingov.first
+    end
+
+    def mhv_credential
+      @mhv_credential ||= user_verifications.mhv.first
+    end
+
+    def dslogon_credential
+      @dslogon_credential ||= user_verifications.dslogon.first
+    end
+
+    def user_verifications
+      @user_verifications ||= user_account.user_verifications
+    end
+
+    def avc_added?
+      user_avc.saved_change_to_acceptable_verified_credential_at?
+    end
+
+    def ivc_added?
+      user_avc.saved_change_to_idme_verified_credential_at?
+    end
+
+    # When the newly added verified_credential_at is the only one that exists and
+    # the user has a mhv credential it is from mhv e.g. user_avc_updater.mhv.{added_type}.added
+    def from_mhv?
```

Comment:
I think for these `from_*` functions we should probably leverage the `user_verification` on the user. This does mean we have to pass in `user_verification` from much higher up, but I think that'll make the logic for all of these much simpler

Reply:
Thats a good call. I'll make that change

Reply:
@bosawt Maybe I am missing something but would we really gain much by passing the `user_verification` ? There would only ever be 2 types of `user_verifications` passed - `idme_uuid`, and `logingov_uuid` which we could infer the type by checking which `verified_credential_at` changed.

Say a user with only a MHV verified account signs in with id.me. 
- A new id.me `user_verification` is created (this is the one we would pass)
-  When we check if their only previous `user_verification` was MHV we would:
   - need to check if they have a MHV `user_verification`
   - check if there is a login.gov `user_verification` (or id.me `user_verification` if the user_verification we pass is logingov)
      - if there is NOT a login.gov `user_verification` it IS a MHV conversion (`user_avc_updater.mhv.ivc.added`)
      - if there IS a login.gov `user_verification` it is NOT a MHV conversion, it would be logingov (`user_avc_updater.logingov.ivc.added`)
      
Something like this:
```ruby
def from_mhv?
  if passed_user_verification.idme_uuid.present?
    mhv_credential.present? && user_verifications.logingov.blank?
  elsif passed_user_verification.logingov.present?
    mhv_credential.present? && user_verifications.idme.blank?
  else
    false
  end
end
```

Versus what it is currently doing where it's checking if the `verified_credential_at` columns changed instead of a `user_verification`
- Checks if MHV credential exists
- Checks if `idme_verified_credential_at` changed and `acceptable_verified_credential_at` is nil, OR  `acceptable_verified_credential_at` changed and `idme_verified_credential_at` is nil.

Currently it is this:
```ruby
def from_mhv?
  if user_avc.saved_change_to_idme_verified_credential_at?
    mhv_credential.present? && user_avc.acceptable_verified_credential_at.nil?
  elsif user_avc.saved_change_to_acceptable_verified_credential_at?
    mhv_credential.present? && user_avc.idme_verified_credential_at.nil?
  else
    false
  end
end
```

Reply:
I think you're right, so I'm gonna retract my previous ask for this. I don't think there's another way to get the kind of information you're trying to log here